### PR TITLE
Modify generator spec to support returning a list of files

### PR DESCRIPTION
**Pull request by mbryzek**
- This is needed to facilitate adding a java client where each class needs to be
   in its own source file.

### CODE REVIEW

File: spec/generator.json [hunk 1]
```diff
@@ -31,10 +31,19 @@
             ]
         },
 
+        "file": {
+            "description": "Represents a source file",
+            "fields": [
+                { "name": "name", "type": "string", "description": "The recommended name for the file, including the path to the file, where appropriate.", "example": "com/myorg/myapplication/Foo.java" },
```
Comment (jkenny) [inline spec/generator.json]: I was thinking of splitting the 'name' field into 'name' and an optional 'path' field. I think this is more self-documenting and easier to consume.

Comment (mbryzek) [inline spec/generator.json]: OK - have added a 'dir' field that is optional. apidoc-generator/master is
fully up to date and ready for release (will plan to release tonight). Note
I've added default file names and directories based on languages (see
https://github.com/mbryzek/apidoc-generator/blob/master/generator/test/lib/ServiceFileNamesSpec.scala
)

apidoc changes should be good to go to:
https://github.com/mbryzek/apidoc/pull/351

If you can take a quick look, can deploy tomorrow.

I've verified that the changes are in fact backwards compatible (using
apidoc-cli). Once released, I'll plan to update apidoc-cli to use the
directory and file names, removing them from the .apidoc config file (not
sure if you use apidoc-cli or not).

And then we should be clear to get the java client merged in.

Mike

Comment (jkenny) [inline spec/generator.json]: My original pull request had refactored the generators to always return a list of files and then the invocations controller was concatenating them into the 'source' field for backwards compatibility. I think we still need that change, right? I can make it as part of the java_models pull request if you want.

Comment (jkenny) [inline spec/generator.json]: Actually disregard my previous comment. The change to the generators is in
a different commit waiting behind this spec change. Thanks!

**Pull request merged**